src/Features/Core/Portable/SemanticSearch/Tools/ReferencingSyntaxFinder.cs

```diff
@@ -34,85 +32,39 @@ public IEnumerable<SyntaxNode> Find(ISymbol symbol)
 
     public async IAsyncEnumerable<SyntaxNode> FindAsync(ISymbol symbol)
     {
-        var channel = Channel.CreateUnbounded<ReferenceLocation>(new()
+        using var _ = PooledHashSet<SyntaxNode>.GetInstance(out var cachedRoots);
```

Comment:
also added this so that we don't continually reparse documents with multiple results in them.